pkg/socket/types.go

```diff
@@ -0,0 +1,30 @@
+package socket
+
+import (
+	"crypto/tls"
+	"time"
+)
+
+type Socket interface {
+	Connect() error
+	Disconnect() error
+	Command(command string, process Processor) error
```

Comment:
Why do we need it now? Do we plan to have several implementations of "Socket"?

Reply:
This with help us with the unit tests once we use the package in a collector. The implementation could just create a mock socket in order to mimic errors rather than implementing a TCP or a UDP server. At the moment in the module we don't need it but it should become handy on the first implementation. I can remove it, but I think it's going to be useful quickly.

Reply:
So, having worked a little bit more on this, I have removed the interface. We can mock the connection in the `Socket` struct for testing. I believe this is also working better with the provided documentation (previous comment).